### ISSUE TITLE
the World.getContainer() call causes major MSPT spikes & lag

### ISSUE THREAD

**Issue by luiskeniosis** [JackFred2/JacksServerSideTweaks#3]
Hi! I installed this on my server running Fabric 0.14.19 for Minecraft 1.19.4 and was noticing some lag after a few days. I installed the "spark" mod and used its profiler to see which mod was causing the lag, and it found that the World.getContainer() used by the worldContainerNames function of your mod was using more than 66% of all the "time per frame". This then combined with Lithium, which it appears to me is updating the lighting around each container was causing some lag.

I wish I had more info but I uninstalled JSST before making this post, but after uninstalling JSST the lag went away completely. Really enjoyed the mod, and if it works fine for others then this is not a big deal, but just at least thought I would let you know.

I can reinstall and provide more info if needed.

**Comment by JackFred2** [comment]
If you still have it, can you send the spark profile (either the link or the downloaded .sparkprofile?) In the meantime, if you still want to use other parts of the mod you can run `/jsst worldContainerNames disable` to disable just that part, which should stop the lag.
![image](https://github.com/JackFred2/JacksServerSideTweaks/assets/8350692/5aa52daa-c235-4cd9-a2e4-2b33c8a0dd04)

**Comment by Merancapeman** [comment]
I can confirm this is still an issue, will try the fix but I had to remove it personally

**Comment by lugui1998** [comment]
Yep, I was going to report this issue and looks like it is already known..
it was a pain to find that this plugin was the cause...

**Comment by venaminecraft** [comment]
Hello @JackFred2, any update on this? 😊

**Comment by JackFred2** [comment]
None right now, I've been planning to rewrite JSST as with the rest of my mods but it's not happened yet.

**Comment by venaminecraft** [comment]
Thanks for the fast answer. Sounds interesting, I keep my eye on it.

**Comment by JackFred2** [comment]
Slightly late; closing this as not planned for the new version ATM. If you want a similar feature I can recommend my other mod Chest Tracker, though it's client-sided.
